environment/src/main/java/com/sequenceiq/environment/environment/service/freeipa/FreeIpaInstanceCountByGroupProvider.java

```diff
@@ -0,0 +1,31 @@
+package com.sequenceiq.environment.environment.service.freeipa;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.environment.api.v1.environment.model.request.AttachedFreeIpaRequest;
+
+@Service
+public class FreeIpaInstanceCountByGroupProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FreeIpaInstanceCountByGroupProvider.class);
+
+    @Value("${environment.freeipa.groupInstanceCount.default}")
+    private Integer defaultInstanceCountByGroup;
+
+    public int getInstanceCount(AttachedFreeIpaRequest request) {
+        int instanceCount = defaultInstanceCountByGroup;
+        if (request != null && request.getInstanceCountByGroup() != null) {
+            instanceCount = request.getInstanceCountByGroup();
```

Comment:
why not just return here with `request.getInstanceCountByGroup()`?

Reply:
I don't like multiple return statements :)